add support for 'AdvertiseAddr' and 'AdvertisePort' in memberlist config

kevinschoonover opened the pull request:
**What this PR does**:
Adds 'advertise_addr' and 'advertise_port' to memberlist configuration to support the service being NATed behind another ip.

Porting PR from cortex (https://github.com/cortexproject/cortex/pull/4387)

**Which issue(s) this PR fixes**:
https://github.com/cortexproject/cortex/issues/4359

**Checklist**
~- [ ] Tests updated~ - unsure how to go about testing this, happy to do so if you could give some pointers
- [X ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

pstibrany commented on CHANGELOG.md:
```diff
@@ -1,5 +1,8 @@
 ## Changelog
 
+* [CHANGE] Memberlist: allow specifying address and port advertised to the memberlist ring by setting the following configuration:
```
Nit: Instead of "memberlist ring", we should say "memberlist cluster members".

(Cortex Ring contains Address field as well, but that isn't affected by these flags.)

pstibrany reviewed (approved):
Thank you. Change looks good, let's please include PR number in the changelog too.

kevinschoonover commented:
Thanks for the review @pstibrany. I addressed your comments, but dont have permission to merge it. Let me know if you have any additional changes you want made!

pstibrany reviewed (approved):
Thank you!
